badparking/mobile_api/views.py

```diff
@@ -39,16 +42,95 @@ class CompleteCurrentUserView(UserObjectMixin, generics.UpdateAPIView):
+            - name: client_id
+              description: ID of the API Client
+              required: true
+              type: string
+              paramType: query
+            - name: client_secret
+              description: Secret of the API Client hashed with a timestamp as `SHA256(secret + timestamp)`
+              required: true
+              type: string
+              paramType: query
+            - name: timestamp
+              description: Current UTC timestamp value in seconds used for hashing the `client_secret`
+              required: true
+              type: integer
+              paramType: query
+            - name: access_token
+              description: Facebook API access token for user authentication
+              required: true
+              type: string
+              paramType: form
+        parameters_strategy:
+            form: replace
+        responseMessages:
+            - code: 201
+              message: User successfully created and authenticated
+            - code: 200
+              message: Matching user found and authenticated
+            - code: 401
+              message: Authentication failed
+        """
+        access_token = request.POST.get('access_token')
+        if not access_token:
+            raise exceptions.ValidationError('Facebook access_token is missing')
+
+        self._get_client(request)
+
+        try:
+            # TODO: add appsecret_proof for better security once facebook-sdk supports it,
+            # see https://github.com/mobolic/facebook-sdk/pull/243
+            graph = facebook.GraphAPI(access_token, version=self.FACEBOOK_API_VERSION)
+            user_info = graph.get_object('me', fields=self.FACEBOOK_USER_FIELDS)
+        except facebook.GraphAPIError as exc:
+            if exc.code in ('OAuthException', '102'):
+                raise exceptions.AuthenticationFailed()
+            else:
+                logger.exception(exc.message)
+                raise exceptions.ValidationError('Facebook access token error')
+
+        user_info = self._map_user_info(user_info)
+        try:
+            user = User.objects.get_by_external_id(user_info['external_id'], user_info.get('email', None))
+            serializer = self.get_serializer(user, data=user_info, partial=True)
+            http_status = status.HTTP_200_OK
+        except User.DoesNotExist:
+            serializer = self.get_serializer(data=user_info)
+            http_status = status.HTTP_201_CREATED
+        serializer.is_valid(raise_exception=True)
+        user = serializer.save()
+
+        return response.Response(serializer.data, status=http_status)
+
+    def _map_user_info(self, user_info):
+        data = {
+            'external_id': user_info['id'],
+            'first_name': user_info.get('first_name', ''),
+            'middle_name': user_info.get('middle_name', ''),
+            'last_name': user_info.get('last_name', '')
+        }
+        if user_info.get('email', None):
```

Comment:
I just like it to be explicit but there's another issue — it should be `if not`. Thanks for this!